packages/typespec-ts/src/utils/operationUtil.ts

```diff
@@ -437,3 +449,133 @@ export function getCollectionFormatHelper(
   const detail = getSpecialSerializeInfo(paramType, paramFormat);
   return detail.descriptions.length > 0 ? detail.descriptions[0] : undefined;
 }
+
+export async function appendDefaultResponseIfAbsent(
+  dpgContext: SdkContext,
+  client: SdkClient
+) {
+  const operationGroups = listOperationGroups(dpgContext, client);
+  const program = dpgContext.program;
+  for (const operationGroup of operationGroups) {
+    const operations = listOperationsInOperationGroup(
+      dpgContext,
+      operationGroup
+    );
+
+    for (const op of operations) {
+      const route = ignoreDiagnostics(getHttpOperationWithCache(program, op));
+      // ignore overload base operation
+      if (route.overloads && route.overloads?.length > 0) {
+        continue;
+      }
+      await appendIfAbsent(route);
+    }
+  }
+  const clientOperations = listOperationsInOperationGroup(dpgContext, client);
+  for (const clientOp of clientOperations) {
+    const route = ignoreDiagnostics(
+      getHttpOperationWithCache(program, clientOp)
+    );
+    if (route.overloads && route.overloads?.length > 0) {
+      continue;
+    }
+    await appendIfAbsent(route);
+  }
+}
+
+async function appendIfAbsent(route: HttpOperation) {
+  const errors = route.responses.filter((r) =>
+    isDefaultStatusCode(r.statusCode)
+  );
+  if (errors.length === 0) {
+    route.responses.push(FAKE_CORE_ERROR_RESPONSE as any);
```

Comment:
There is no easy way to append the default error response so I  build a fake model to use if absent.